Use "filled" bookmark icon when the item is bookmarked

nemanjaglumac opened the pull request:
This PR resolves #34327 by increasing the icon size, and by changing the icon type to filled once it is selected.

Take a look at the original issue > Additional context to get a sense for how this should look.

These are the screenshots from this branch:
![image](https://github.com/metabase/metabase/assets/31325167/4151b1b3-39c1-4cd5-be0b-e4e40c61ca17)
![image](https://github.com/metabase/metabase/assets/31325167/06bd9120-fed7-4890-aef0-ef50ebbbbc1d)

![image](https://github.com/metabase/metabase/assets/31325167/44de6662-4f6e-4e5f-99f9-bbea812f8856)
![image](https://github.com/metabase/metabase/assets/31325167/d59bb092-0009-488d-ad70-fde7b4053fab)

nemanjaglumac committed:
Use "filled" bookmark icon when the item is bookmarked

nemanjaglumac committed:
Update E2E tests

nemanjaglumac committed:
Update the affected unit test

mazameli reviewed (changes requested):
Thank you for doing this!

Looks like the icon isn't quite centered within this hover state though: 

<img width="388" alt="image" src="https://github.com/metabase/metabase/assets/2223916/e95a379f-5531-42a6-815a-1d5e3cdc2039">

WiNloSt commented on frontend/src/metabase/core/components/BookmarkToggle/BookmarkToggle.tsx:
```diff
@@ -37,6 +37,10 @@ const BookmarkToggle = forwardRef(function BookmarkToggle(
     setIsAnimating(false);
   }, []);
 
+  const getIcon = () => {
+    return isBookmarked ? "bookmark_filled" : "bookmark";
+  };
```
I'd say it's more practical just to make this a variable i.e.

```suggestion
  const icon = isBookmarked ? "bookmark_filled" : "bookmark";
```

This is because functions are supposed to be called by others with some of these reasons
1. The function is called at a different time than its instantiation.
2. You expect the caller to pass some arguments to the function.
3. You don't want the value to be evaluated when the function is being declared e.g. you could reference a variable that's being instantiated after the declaration of a function. Because the function will be called after the variable is instantiated, it would have value. This wouldn't work if you evaluated the variable right away.

There could be more reasons, but these are some reasons that are perfectly valid for putting things in a function. Other than that, if both functions and variables work, I'd prefer the variables first. Like in this case, you could think of it like this. Why do I need a function when I'll immediately call it immediately below its declaration?

WiNloSt commented on frontend/src/metabase/nav/containers/MainNavbar/MainNavbarContainer/BookmarkList/BookmarkList.tsx:
```diff
@@ -84,6 +84,8 @@ const BookmarkItem = ({
     bookmark.type === "collection" &&
     !PLUGIN_COLLECTIONS.isRegularCollection(bookmark);
 
+  const getIcon = () => (isSelected ? "bookmark_filled" : "bookmark");
```
Same as the previous comment. Let's make this a variable. Also please note the different style of the functions. The previous one is using `return` while this one is using a shorthand return. Mostly we'd want things to be consistent. It's doesn't matter in this case if we would change it to a variable anyway, but that's something to keep in mind.

nemanjaglumac commented:
> Two other things I noticed:
> 
> 1. My figma mock incorrectly assumed we were using `text-dark` for the action icons, but it looks like in prod we're using `text-medium`, so the bookmark one which is using `text-dark` in this branch now looks darker than the others.
> 2. Something else that I didn't call out is that we're using 20px icons for the action buttons on dashboards, but aren't for some reason on questions. I think we should be, which would then make the actions on saved questions match the size of the bookmark action. Here's the current state of things, with dashboard actions on the bottom, where you can see how they're larger.
> 
> <img alt="image" width="576" src="https://private-user-images.githubusercontent.com/2223916/300923510-1dc0f816-37d1-48ef-82a4-b4eff4c2c0cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY5MDE0NDIsIm5iZiI6MTcwNjkwMTE0MiwicGF0aCI6Ii8yMjIzOTE2LzMwMDkyMzUxMC0xZGMwZjgxNi0zN2QxLTQ4ZWYtODJhNC1iNGVmZjRjMmMwY2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIwMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMDJUMTkxMjIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGIwZTVmZTI4NTdkMmE0OTNiM2ZjMDU0MzNiOTVjNmZjYTFhMTk0Y2MyODU5N2ZlZDk2MTQ1ZDA5Mzk2Njk3MCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.g9kygef4cPZ3IGei39iP1l7RdGpgSzaHgJDaH0PNM5w">
> cc @kdoh just so he's aware of these changes.

@maz, now that you mentioned it, I found even more differences.

- dashboard icons use `text-medium`, question icons use `text-dark`
- dashboard icons' size is 20, question icons' size is 16

Which color should we use?

And something I've also just noticed - bookmark icon is the only one that doesn't change its color to `color-brand` on hover. Would you like me to fix that as well?

mazameli commented:
> dashboard icons use text-medium, question icons use text-dark
dashboard icons' size is 20, question icons' size is 16
Which color should we use?

@kdoh and I spoke, and let's go with 16px, `text-dark` for both dashboards and saved questions.

>And something I've also just noticed - bookmark icon is the only one that doesn't change its color to color-brand on hover. Would you like me to fix that as well?

Yes please! (I think we weren't doing that before, because before the only indication of whether an item was bookmarked or not was if it was blue or not.)

nemanjaglumac committed:
Make dashboard header icons smaller and darker

nemanjaglumac committed:
Make dashboard bookmark icon look like its neighbors

nemanjaglumac commented:
@mazameli Thank you for the input. I have addressed all your comments. Things look more unified now. Please give this one final look.

I have found some more inconsistencies that are not specific to the bookmark icon, but that's a broader topic I'm happy to chat about. That falls outside the scope of this particular PR, imo.

@WiNloSt addressed your comments as well. Please check once again.

nemanjaglumac committed:
Unify question and dashboard icon's background on hover

Question icons have been using `text-medium` unlike dashboard icons
that had `text-light`.

WiNloSt reviewed (approved):
Code-wise. Looks good to me 👍

mazameli reviewed (approved):
Thank you!

nemanjaglumac commented:
@metabase-bot backport release-x.48.x
